applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp

```diff
@@ -0,0 +1,276 @@
+            const double finite_diff_sensitivity =
+                -(response_value1 - response_value0) / delta;
+            const double adjoint_sensitivity =
+                adjoint_solver.CalculateSensitivity(i_node, dir);
+            KRATOS_CHECK_NEAR(adjoint_sensitivity, finite_diff_sensitivity, 1e-9);
+        }
+    }
+}
+} // namespace Testing
+
+namespace // cpp internals
+{
+namespace smtsts
+{ // cotire unity guard
+typedef TUblasSparseSpace<double> SparseSpaceType;
+typedef TUblasDenseSpace<double> LocalSpaceType;
+typedef LinearSolver<SparseSpaceType, LocalSpaceType> LinearSolverType;
+typedef Scheme<SparseSpaceType, LocalSpaceType> SchemeType;
+typedef ConvergenceCriteria<SparseSpaceType, LocalSpaceType> ConvergenceCriteriaType;
+typedef SolvingStrategy<SparseSpaceType, LocalSpaceType, LinearSolverType> SolvingStrategyType;
+
+AdjointResponseFunction::Pointer ResponseFunctionFactory(ModelPart* pModelPart, unsigned ResponseNodeId)
+{
+    Parameters params{R"({"traced_dof": "DISPLACEMENT_Y", "gradient_mode": "semi_analytic", "step_size": 1e-2})"};
+    params.AddEmptyValue("traced_node_id");
+    params["traced_node_id"].SetInt(ResponseNodeId);
+    return Kratos::make_shared<AdjointNodalDisplacementResponseFunction>(*pModelPart, params);
+}
+
+SolvingStrategyType::Pointer CreatePrimalSolvingStrategy(ModelPart* pModelPart)
+{
+    LinearSolverType::Pointer p_linear_solver =
+        Kratos::make_shared<SkylineLUCustomScalarSolver<SparseSpaceType, LocalSpaceType>>();
+    SchemeType::Pointer p_scheme =
+        Kratos::make_shared<ResidualBasedBossakDisplacementScheme<SparseSpaceType, LocalSpaceType>>(0.0);
+    ConvergenceCriteriaType::Pointer p_conv_criteria =
+        Kratos::make_shared<ResidualCriteria<SparseSpaceType, LocalSpaceType>>(1e-24, 1e-25);
+    return Kratos::make_shared<ResidualBasedNewtonRaphsonStrategy<SparseSpaceType, LocalSpaceType, LinearSolverType>>(
+        *pModelPart, p_scheme, p_linear_solver, p_conv_criteria, 30, true, false, true);
+}
+
+void SolvePrimal(ModelPart* pModelPart,
+                 double DeltaTime,
+                 double TotalTime,
+                 std::function<void(ModelPart*)> CallerFun)
+{
+    auto p_solver = CreatePrimalSolvingStrategy(pModelPart);
+    p_solver->Initialize();
+    const double start_time = 0.0;
+    pModelPart->CloneTimeStep(start_time - DeltaTime);
+    pModelPart->CloneTimeStep(start_time);
+    for (double current_time = start_time; current_time < TotalTime;)
+    {
+        current_time += DeltaTime;
+        pModelPart->CloneTimeStep(current_time);
+        p_solver->Solve();
+        CallerFun(pModelPart);
+    }
+}
+
+double CalculateResponseValue(ModelPart* pModelPart, double DeltaTime, double TotalTime, unsigned ResponseNodeId)
+{
+    auto p_response_function = ResponseFunctionFactory(pModelPart, ResponseNodeId);
+    p_response_function->Initialize();
+    double response_value = 0.;
+    auto sum_response_value = [&](ModelPart* pModelPart) {
```

Comment:
Is such transient response value calculation realized in python (e.g. in the adjoint fluid)?

Reply:
No, only here.